diff --git a/smi_main.c b/smi_main.c
index 118a508..f668a46 100644
--- a/smi_main.c
+++ b/smi_main.c
@@ -42,6 +42,9 @@ static int smi_user_framebuffer_dirty(struct drm_framebuffer *fb, struct drm_fil

 static struct drm_framebuffer *smi_user_framebuffer_create(struct drm_device *dev,
 							   struct drm_file *filp,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 17, 0)
+							   const struct drm_format_info *info,
+#endif
 							   const struct drm_mode_fb_cmd2 *mode_cmd);

 static const struct drm_framebuffer_funcs smi_fb_funcs = {
@@ -148,9 +151,11 @@ static int smi_handle_damage(struct drm_framebuffer *fb, struct drm_clip_rect cl

 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
 	gbo = drm_gem_vram_of_gem(obj);
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(6, 17, 0)
 	ret = drm_gem_vram_pin(gbo, DRM_GEM_VRAM_PL_FLAG_VRAM);
 	if (ret)
 		return (0);
+#endif

 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
 	ret = drm_gem_vram_vmap(gbo, &dst_map);
@@ -232,7 +237,9 @@ static int smi_handle_damage(struct drm_framebuffer *fb, struct drm_clip_rect cl
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
 	if (kmap)
 	{
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(6, 17, 0)
 		drm_gem_vram_unpin(gbo);
+#endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
 		drm_gem_vram_vunmap(gbo, &dst_map);
 #elif LINUX_VERSION_CODE >= KERNEL_VERSION(5, 9, 0)
@@ -300,9 +307,16 @@ unlock:

 static struct drm_framebuffer *smi_user_framebuffer_create(struct drm_device *dev,
 							   struct drm_file *filp,
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 17, 0)
+							   const struct drm_format_info *info,
+#endif
 							   const struct drm_mode_fb_cmd2 *mode_cmd)
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 17, 0)
+	return (drm_gem_fb_create_with_funcs(dev, filp, info, mode_cmd, &smi_fb_funcs));
+#else
 	return (drm_gem_fb_create_with_funcs(dev, filp, mode_cmd, &smi_fb_funcs));
+#endif
 }

 /*
diff --git a/smi_plane.c b/smi_plane.c
index f64fc54..210139d 100644
--- a/smi_plane.c
+++ b/smi_plane.c
@@ -135,15 +135,19 @@ static void smi_cursor_atomic_update(struct drm_plane *plane,
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
 		gbo = drm_gem_vram_of_gem(fb->obj[0]);

+#if LINUX_VERSION_CODE <= KERNEL_VERSION(6, 17, 0)
 		if (drm_gem_vram_pin(gbo, DRM_GEM_VRAM_PL_FLAG_VRAM) < 0) {
 			dbg_msg("vram_pin failed\n");
 			LEAVE();
 		}
+#endif

 		cursor_offset = drm_gem_vram_offset(gbo);
 		if (cursor_offset < 0) {
 			dbg_msg("get gpu_addr failed\n");
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(6, 17, 0)
 			drm_gem_vram_unpin(gbo);
+#endif
 			LEAVE();
 		}
 		if (sdev->specId == SPC_SM750) {
@@ -184,7 +188,9 @@ static void smi_cursor_atomic_update(struct drm_plane *plane,
 					  BPP32_BLUE);
 			ddk768_enableCursor(disp_ctrl, 3);
 		}
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(6, 17, 0)
 		drm_gem_vram_unpin(gbo);
+#endif
 #else
 		bo = gem_to_smi_bo(fb->obj[0]);
 		if (smi_bo_pin(bo, TTM_PL_FLAG_VRAM, &cursor_offset) < 0) {
@@ -304,9 +310,11 @@ __attribute__((unused)) static int smi_primary_plane_prepare_fb(struct drm_plane
 			continue;
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
 		gbo = drm_gem_vram_of_gem(new_state->fb->obj[i]);
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(6, 17, 0)
 		ret = drm_gem_vram_pin(gbo, DRM_GEM_VRAM_PL_FLAG_VRAM);
 		if (ret < 0)
 			goto err_drm_gem_vram_unpin;
+#endif
 #else
 		bo = gem_to_smi_bo(new_state->fb->obj[i]);
 		ret = smi_bo_pin(bo, TTM_PL_FLAG_VRAM, NULL);
@@ -318,6 +326,7 @@ __attribute__((unused)) static int smi_primary_plane_prepare_fb(struct drm_plane
 	LEAVE(0);

 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(6, 17, 0)
 err_drm_gem_vram_unpin:
 	while (i) {
 		--i;
@@ -325,6 +334,7 @@ err_drm_gem_vram_unpin:
 		gbo = drm_gem_vram_of_gem(new_state->fb->obj[i]);
 		drm_gem_vram_unpin(gbo);
 	}
+#endif
 #else
 err_smi_bo_unpin:
 	while (i) {
@@ -355,7 +365,9 @@ __attribute__((unused)) static void smi_primary_plane_cleanup_fb(struct drm_plan
 			continue;
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 3, 0)
 		gbo = drm_gem_vram_of_gem(old_state->fb->obj[i]);
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(6, 17, 0)
 		drm_gem_vram_unpin(gbo);
+#endif
 #else
 		bo = gem_to_smi_bo(old_state->fb->obj[i]);
 		smi_bo_unpin(bo);
